pallets/compliance-manager/src/lib.rs

```diff
@@ -735,9 +832,22 @@ impl<T: Trait> Module<T> {
                 asset_compliance_with_results.result = true;
             }
         }
+        // Aggregating the result with the implicit requirement result.
+        asset_compliance_with_results.result =
+            Self::compute_final_result(&asset_compliance_with_results);
```

Comment:
Instead of computing the final result again, It will be easier to understand if you added a check in the original line that sets the final result. i.e. the following line
```
if requirements.result {
    asset_compliance_with_results.result = true;
}
```
can become something like
```
if requirements.result && implicit_result {
    asset_compliance_with_results.result = true;
}
```

Alternatively, you can cache the requirements result in a different value and then use && on caches values to calculate the final result.